models/bamf_nnunet_ct_lung/utils/LungPostProcessor.py

```diff
@@ -0,0 +1,81 @@
+"""
+-------------------------------------------------
+MHub - Run Module for perform postprocessing logic on segmentations.
+-------------------------------------------------
+-------------------------------------------------
+Author: Jithendra Kumar
+Email:  jithendra.kumar@bamfhealth.com
+-------------------------------------------------
+"""
+from mhubio.core import IO
+from mhubio.core import Module, Instance, InstanceData
+import os, shutil
+import SimpleITK as sitk
+import numpy as np
+from skimage import measure, filters
+
+
+class LungPostProcessor(Module):
+
+    def get_lung(self, ip_path):
+        # get lung segmentation
+        seg_data = sitk.GetArrayFromImage(sitk.ReadImage(ip_path))
+        seg_data[seg_data > 0] = 1
+        return seg_data
+
+    def n_connected(self, img_data):
+        img_data_mask = np.zeros(img_data.shape)
+        img_data_mask[img_data >= 1] = 1
+        img_filtered = np.zeros(img_data_mask.shape)
+        blobs_labels = measure.label(img_data_mask, background=0)
+        lbl, counts = np.unique(blobs_labels, return_counts=True)
+        lbl_dict = {}
+        for i, j in zip(lbl, counts):
+            lbl_dict[i] = j
+        sorted_dict = dict(sorted(lbl_dict.items(), key=lambda x: x[1], reverse=True))
+        count = 0
+
+        for key, value in sorted_dict.items():
+            if count >= 1 and count <= 2:
+                print(key, value)
+                img_filtered[blobs_labels == key] = 1
+            count += 1
+
+        img_data[img_filtered != 1] = 0
+        return img_data
+
+    def get_seg_img(self, lungs, nodules, ct_path):
```

Comment:
It would be nice if you could add a short description of what this function does and how it operates as this step is specific to the pipeline.

Reply:
I have added short description to the functions.

Reply:
thank you!